Atheer_Minwar/week_09/day_01/weatherApp/app/javascript/components/App.js

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import Search from "./Search";
+import SearchResult from "./SearchResult";
+
+class App extends React.Component {
+  constructor() {
+    super();
+    this.state = { data: null };
+    this.search = this.search.bind(this);
+  }
+
+  search(term) {
+    fetch(
+      `api.openweathermap.org/data/2.5/weather?q=${term}&APPID=e36837de9b3cb9a17e5d4d35fa20707a`
+    )
+      .then(respo => respo.json())
+      .then(data => {
+        this.setState({
+          data: data
+        });
+      });
+  }
+
+  render() {
+    const { coord, weather, sys } = this.state.data;
```

Comment:
So the reason you're getting the `can't read data of property null` is because here you're trying to get the data before the `fetch` line is actually done (promise hasn't yet been fulfilled). One way to get around that is to have a condition/if block and render different things based on the status of the state, it can look like something:
```js
if(data === null){
   return(
    // loading indication of some sort
   );
}

const { coord, weather, sys } = this.state.data;
return(
  // actual data
);
```

Hope this helps!